ilastik/widgets/filePathButton.py

```diff
@@ -1,91 +1,82 @@
-        self.setToolTip(self.text())
-
-    def resizeEvent(self, event):
-        # Start with the full path
-        short_filepath = self._filepath
-        self.setText(self._filepath + self._suffix)
-        ideal_width = self.minimumSizeHint().width()
-
-        # Keep removing intermediate directories until the text fits inside the new width
-        while event.size().width() < ideal_width:
-            drive, short_filepath = os.path.splitdrive(short_filepath)
-            dirpath, filename = os.path.split(short_filepath)
-
-            dir_names = dirpath.split(os.sep)[1:]
-            if len(dir_names) == 0 or (len(dir_names) == 1 and dir_names[0] == "..."):
-                # This path is already as short as possible.  Give up.
-                return
-
-            if dir_names and dir_names[0] == "...":
-                dir_names = dir_names[1:]
-            dir_names = [""] + ["..."] + dir_names[1:]
-            dirpath = os.sep.join(dir_names)
-
-            # Always include drive
-            short_filepath = drive + os.path.join(dirpath, filename)
-            self.setText(short_filepath + self._suffix)
-            ideal_width = self.minimumSizeHint().width()
-
-
-if __name__ == "__main__":
-    from PyQt5.QtWidgets import QApplication, QWidget, QVBoxLayout
-
-    app = QApplication([])
-
-    SIMULATE_WINDOWS = True
-    if SIMULATE_WINDOWS:
-        import ntpath
-
-        os.sep = ntpath.sep
-        os.path = ntpath
-
-        buttons = []
-        buttons.append(FilePathButton(r"c:\some\long\path\to\the\file.txt", " (Don't abbreviate this suffix)"))
-        buttons.append(FilePathButton(r"file.txt", " (some suffix)"))
-        buttons.append(FilePathButton("\\some\\long\\path\\to\\the\\", " (some suffix)"))
-    else:
-        buttons = []
-        buttons.append(FilePathButton("/some/long/path/to/the/file.txt", " (Don't abbreviate this suffix)"))
-        buttons.append(FilePathButton("file.txt", " (some suffix)"))
-        buttons.append(FilePathButton("/some/long/path/to/the/", " (some suffix)"))
-
-    layout = QVBoxLayout()
-    for button in buttons:
-        # Typically, the button chooses a minimum size that is appropriate for its shortest possible text
-        # But for this little test we force an even smaller minimum size so we can test multiple paths at once...
-        button.setMinimumWidth(10)
-        layout.addWidget(button)
-
-    widget = QWidget()
-    widget.setLayout(layout)
-    widget.show()
-
-    app.exec_()
+        super().__init__(parent)
+        self.setSizePolicy(QSizePolicy.Preferred, QSizePolicy.Fixed)
+
+        self._path = PurePath(path)
+        self.setToolTip(str(self._path))
+
+        self._suffix = f"\n{subtext}" if subtext else ""
+
+        # Icon should be set before the text in order to be accounted for in the size hint.
+        if icon is not None:
+            super().setIcon(icon)
+
+        self._abbrevs = [(PurePath(), 0)]
```

Comment:
any reason to initialize `self._abbrevs` here, also with a value type (`tuple`) that is otherwise not used in this variable.

```suggestion
```

Reply:
This is just to always have an initialized field in `__init__` with some default value.

Reply:
Okay, then it should probably go into the else of the above if clause, as this already calls `_update_abbrevs` if there is an icon.

Reply:
It calls `setIcon()` from `super()`, so it just sets an icon.

i agree that this class doesn't behave in the most intuitive way, but I haven't been able to come up with a better solution (`resizeEvent` needs to be fast, so we can't recompute `_abbrevs` every time there).

Reply:
ah right, I'll not loose sleep over it.